[1.x] Enable multipart/related on FileUpload

mufasa1976 opened the pull request:
SOAP-Requests with MTOM enabled are sending the Requests with Content-Type `multipart/related`.

Because this Project is used by [Wiremock](https://github.com/wiremock/wiremock) it would fix the Issue [#2176: "multipart/related request matching fails if the content-disposition header is not present"](https://github.com/wiremock/wiremock/issues/2176)

jochenw commented:
Would you, please, complete your PR, by

- Adding yourself as a contributor in pom.xml.
- Adding an entry to src/changes/changes.xml
- Rebase it to HEAD

Thanks,

Jochen

mufasa1976 commented:
2 of 3 I can do:
- register as a Contributor in pom.xml
- make a new pull-request to rebase the changes to the master

But I don't know how to add the Entry to the src/changes/changes.xml because there is no new Version 1.6 and I don't know the Release-Date. Would you please support me doing this?

garydgregory commented on src/main/java/org/apache/commons/fileupload/FileUploadBase.java:
```diff
@@ -138,6 +138,11 @@ public static boolean isMultipartContent(final HttpServletRequest req) {
+    /**
+     * HTTP content type header for multiple related data.
+     */
+    public static final String MULTIPART_RELATED = "multipart/related";
```
New public and protected items need a Javadoc 'since tag.

mufasa1976 replied:
I've added a `@since 1.6` comment. Hope this is correct

garydgregory commented on src/main/java/org/apache/commons/fileupload/FileUploadBase.java:
```diff
@@ -1069,13 +1080,23 @@ private boolean findNextItem() throws IOException {
                         eof = true;
                         return false;
                     }
+
```
No extra whitespace needed.

garydgregory commented on src/test/java/org/apache/commons/fileupload/FileUploadTest.java:
```diff
@@ -396,4 +397,53 @@ private void assertHeaders(final String[] pHeaderNames, final String[] pHeaderVa
+
+    /**
+     * Test for multipart/related without any content-disposition Header.
+     * <p/>
```
No empty tag, instead enclose the paragraph.

mufasa1976 replied:
I've removed the Paragraph

garydgregory reviewed (changes requested):
Hello @mufasa1976
Thank you for your PR.
Please see my comments.

jochenw commented:
Just noticed: Why is the destination 1.x, and not master, as it should be? The 1.x branch is in maintenance mode: We try very hard, to leave it as it is.

mufasa1976 commented:
> Just noticed: Why is the destination 1.x, and not master, as it should be? The 1.x branch is in maintenance mode: We try very hard, to leave it as it is.

This is easy to explain. I have to test SOAP Requests with MTOM-Attachments. With the Standard SAAJ-Implementation enabled this will just send `multipart/related` Requests.

The Standard Test-Implementation of Spring (spring-ws-test) only supports easy SOAP-Requests without MTOM Requests (which are using `application/soap+xml` as content-type).

That's why I changed to the Project [Wiremock](https://github.com/wiremock/wiremock) to test for `multipart/related` SOAP-Requests.

Because Wiremock at the Moment uses commons-fileupload Version 1.5 I fixed this Issue on the `1.x` Branch and prepared the same Fix as a separate Pull-Request for the master-Branch of commons-fileupload (for Version 2.x upwards).

I don't know when it is planned that the Project Wiremock will use Version 2.x of commons-fileupload. That's the Reason why I fixed it twice (one time for the `1.x` Branch, second Time for the `2.x` Branch).

But I will wait for the PR of the `2.x` Branch until this PR has been merged (hopefully there will be a 1.6 Version).

garydgregory reviewed (changes requested):
Hello @mufasa1976 
Thank you for your updates.
Please rebase this PR on Git master as this will allow your PR to run builds and checks.
TY!

mufasa1976 commented:
I've just created a new PR for the master-Branch: https://github.com/apache/commons-fileupload/pull/315
